### PR TITLE
Support pyqt6 and python 3.10

### PR DESCRIPTION
**Pull request by StFroese**
Unfortunately ViTables wasn't working on M1 Macs because PyQT5 wasn't installable. 
I added PyQT6 and it seems to work no with python 3.10

### CODE REVIEW

**Comment by StFroese** [comment]
@uvemas pls review :)

File: setup.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def read(fname):
     return open(os.path.join(os.path.dirname(__file__), fname)).read()
 
 
-install_requires = ['qtpy (>=1.2.1)', 'numpy (>=1.4.1)', 'numexpr (>=2.0)',
+install_requires = ['qtpy', 'pyqt6', 'numpy (>=1.4.1)', 'numexpr (>=2.0)',
                     'tables (>=3.0)', 'setuptools']
 if 'CONDA_PREFIX' not in os.environ:
     install_requires.append('PyQt5 (>=5.5.1)')
```
Comment (StFroese) [inline setup.py:35]: Maybe this also needs to be changed to PyQT6?

**Comment by StFroese** [comment]
Should also work with python 3.11

**Comment by uvemas** [comment]
Thanks a lot for your contrib @StFroese .

**Comment by StFroese** [comment]
@uvemas no problem :)
Can we do a new release v3.0.3? I would then do a new conda release :)

**Comment by uvemas** [comment]
Sorry, I know that a new release is needed and that my project maintenance is really poor. But for now, it's all I can do, and I don't see this situation changing in the short term. Anyway, I'll try to be more active and work on a new release.

Thanks again for your help!

**Comment by StFroese** [comment]
@uvemas I can also do the new release as an "as it is right now version". So would 3.0.3 would basically be support for python 3.10 and 3.11 as well as support for M1 Macs. I can open a PR, you just have to review :)

**Comment by JaRoSchm** [comment]
After my recent pull request to qtpy (https://github.com/spyder-ide/qtpy/pull/440) this would also enable compatibility with PySide6 (new qtpy release needed too).

**Comment by uvemas** [comment]
Great news!

**Pull request merged**